### PR TITLE
Add FastAPI and SQLModel to third-party tests

### PR DESCRIPTION
**Pull request by sydney-runkle**
Part of https://github.com/pydantic/pydantic/issues/11160.

### CODE REVIEW

**Review by sydney-runkle — COMMENTED** [review]
Looks great, thanks!

Screenshot 2024-12-20 at 10 38 48 AM

Note, re screenshot above - not sure if we need the ticks as it doesn't display as code on the actions list or actions drilldown.

Also, before we merge this PR, could we add a section to `contributing.md` that explains our criteria / process for adding integration tests like this? Ex - # of starts, amount of usage, etc?

File: docs/contributing.md [hunk 1]
```diff
@@ -253,3 +254,19 @@ Pydantic has a badge that you can use to show that your project uses Pydantic. Y
 
 <a href="https://pydantic.dev"><img src="https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/pydantic/pydantic/main/docs/badge/v2.json" alt="Pydantic Version 2" style="max-width:100%;"></a>
 ```
+
+## Adding your library as part of Pydantic tests
```
Comment (sydney-runkle) [inline docs/contributing.md:258]:
```suggestion
## Adding your library as part of Pydantic's third party test suite
```

File: docs/contributing.md [hunk 2]
```diff
@@ -253,3 +254,19 @@ Pydantic has a badge that you can use to show that your project uses Pydantic. Y
...
+
+To be able to identify regressions early during development, Pydantic runs tests on various third-party projects
+using Pydantic. Are you maintaining an open source project and are you heavily relying on Pydantic? If you match
+some of the following criteria:
```
Comment (sydney-runkle) [inline docs/contributing.md:262]:
```suggestion
using Pydantic. We consider adding support for testing new open source projects (that rely heavily on Pydantic) if your said project matches some of the following criteria:
```

File: docs/contributing.md [hunk 3]
```diff
@@ -253,3 +254,19 @@ Pydantic has a badge that you can use to show that your project uses Pydantic. Y
...
+
+- The project is actively maintained.
+- Making use of Pydantic internals (e.g. relying on the [`BaseModel`][pydantic.BaseModel] metaclass, typing utilities).
```
Comment (sydney-runkle) [inline docs/contributing.md:265]:
```suggestion
- The project makes use of Pydantic internals (e.g. relying on the [`BaseModel`][pydantic.BaseModel] metaclass, typing utilities).
```

File: docs/contributing.md [hunk 4]
```diff
@@ -253,3 +254,19 @@ Pydantic has a badge that you can use to show that your project uses Pydantic. Y
...
+- The project is popular enough (although small projects can still be included depending on how Pydantic is being used).
```
Comment (sydney-runkle) [inline docs/contributing.md:266]: I feel like we should quantify this (some # of stars, downloads)

File: docs/contributing.md [hunk 5]
```diff
@@ -253,3 +254,19 @@ Pydantic has a badge that you can use to show that your project uses Pydantic. Y
...
+- The project CI is simple enough to be ported into Pydantic's workflow.
```
Comment (sydney-runkle) [inline docs/contributing.md:267]:
```suggestion
- The project CI is simple enough to be ported into Pydantic's testing workflow.
```

File: docs/contributing.md [hunk 6]
```diff
@@ -253,3 +254,19 @@ Pydantic has a badge that you can use to show that your project uses Pydantic. Y
...
+
+The project can be tested as part of Pydantic's development process. You can [open feature request][open feature request]
+to discuss the inclusion of your project.
```
Comment (sydney-runkle) [inline docs/contributing.md:270]:
```suggestion
If your project meets some of these criteria, you can [open feature request][open feature request]
to discuss the inclusion of your project.
```

Or something like that, feel free to workshop, seems a little redundant right now with the above leading sentence

**Review by sydney-runkle — COMMENTED** [review]
Looks good, just a bit of workshoping on wording.

File: docs/contributing.md (hunk 4 above)
Comment (Viicos) [reply docs/contributing.md:266]: The thing is it really depends on how Pydantic is used. We can have project with 100k stars but if it is only defining simple models (e.g. models for an API client, and types are well contained) then there's no interest in testing it. On the other hand, a project with 200 stars defining complex custom types is more valuable. So it is hard to quantify :/

**Review by Viicos — APPROVED** [review]

**Pull request merged**